Develop

bachynskyi opened the pull request.

Commit:
header, route

Додано адаптивний заголовок, додано   модуль для роутів, додано всі картинки.

Commit:
Робота з header, додано компонент not-found

Поправив шрифти і розмір в header. Додано нову компоненту для відсутнього адресу.

Commit:
Sidenav, Route

Виправив помилку Sidenav при якій при виклику Sidenav не затінялись всі сектори, виприв помилку при якій перейшовши за будь яким адресом Route редіректив на home

Commit:
not-found

зверстана сторінка 404

bachynskyi commented on src/app/home/home.component.ts:
```diff
@@ -0,0 +1,15 @@
+  constructor() { }
+
+  ngOnInit() {
+  }
```
Бачу що в тебе багато пустих функцій, Якщо ти їх не використовуєш, то краще видалити

bachynskyi replied:
`constructor` також краще видаляти, якщо він пустий

Alexander1998007 replied:
просто я створюю всі компоненти через термінал і він сам їх прописує, потім як закінчу то почищу весь код, бо там і в scss тоже багато мусору.

bachynskyi commented on src/app/navigation/sidenav-list/sidenav-list.component.scss:
```diff
@@ -0,0 +1,23 @@
+ 
+.nav-caption{
+    display: inline-block;
+    padding-left: 6px;
```
Тут варто додати властивість `text-transform`

Alexander1998007 replied:
до нього ще не добрався, як закінчу з каруселсю то зразу візьмусь за нього, хочу ще його стилізувати відповідно до сторінки

Commit:
header

Готовий header

Commit:
search component

Додано шрифти, частково зроблена сторінка пошуку

Commit:
search

Додав select, кнопки "All categories" і "How it works?". Select потрібно доопрацювати.

Commit:
product component

Майже готово. Зверстано всі елементи. Доопрацювати відображення на мобільних телефонах.

Commit:
footer component

Додано і зверстано footer.

Commit:
bottom-menu component, carousel component

Додано і зверстано компоненту bottom-menu. Додано компоненту carousel і зверстано стрічки. В компоненті footer поправлено правильне відображення стрічки.

Commit:
carousel component

Додано прототип каруселі.

Commit:
carousel component

Додано стиль для регулювання активних і центального елементу каруселі

bachynskyi commented on src/app/bottom-menu/bottom-menu.component.scss:
```diff
@@ -0,0 +1,46 @@
+  font-family: 'RalewayMedium';
+  text-decoration: none;
+  color: #f7f7f7;
+  font-size: 1.5vh; //розмір шрифта
```
Ти ще не міняв розміри шрифтів і педінги?

Alexander1998007 replied:
зараз цим займаюсь, роблю все по порядкую, почав з search компоненти, вже зробив, зараз буду переробляти product компоненту

bachynskyi commented on src/app/search/search.component.html:
```diff
@@ -0,0 +1,41 @@
+      INNOVATE
+    </div>
+    <div class="flex-item">
+      • LIST •
```
Точки так не треба робити. Краще через css

bachynskyi replied:
Також треба додати text-transform

Alexander1998007 replied:
поняв, вже зробив

Commit:
search component

Поправив responsive, зробив через flexbox.

Commit:
search component

Виправив написання кнопки через scss, використав text-transform

Commit:
products component

Поправив responsive, зробив через flexbox.

Commit:
footer component

Поправив responsive, зробив через flexbox.

Commit:
bottom-menu component

Поправив responsive, зробив через flexbox.

Commit:
header, sidenav-list, carousel components

Додав іконки і text-transform в sidenav-list; Додав text-transform в header;   Поправив responsive, зробив через flexbox у carousel.

Commit:
not-found component

Зробив перехід на головну сторінку без перезагрузки сторінки.

Commit:
fonts-style, select in search component

Додав fonts-style.scss який містить всі стилі шрифтів; Select в search component зробив з використанням angular material.

Commit:
search component

Додав text-transform.

Commit:
products component, fonts, select

Виправив всі розмірність всіх шрифтів; Виправив select так щоб там не було нічого лишнього; Зробив відображення максимум 3 товарів в рядку у products component.

Commit:
header component

Зробив зміну кольору хедера на scroll

Commit:
carousel, task-js component

Замінив у carousel компоненті бібліотеку owl на бібліотеку swiper; Додав нову компоненту task-js для роботи з JSON файлами; Налаштував tsconfig.json для підтримки json файлів; Налаштував routing.module для використання task-js компоненти; Додав картинки для JSON.

Commit:
task-js component

зробив відображення для кожної категорії відповідних items

Commit:
task-js component

Зробив відображення картинок

Commit:
task-js component

Виправив стрічку підключення картинок

Commit:
task-js component

Додав кнопку на відображення item без категорій і з категоріями; До картинок додав flex.

Commit:
task-js

Поправив flex для items без категорій.

Commit:
task-js, carousel component

Поправив відображення картинок у task-js. Зациклив карусель, додав відступи між слайдами.

bachynskyi commented on src/app/navigation/header/header.component.scss:
```diff
@@ -2,7 +2,7 @@ a {
   text-decoration: none;
   color: black;
-  font-size: 10px; //розмір шрифта в хедері
+  font-size: 0.5em; //розмір шрифта в хедері
```
якщо в тебе в html стоїть `font-size: 100%`, то означає що `1rem = 16px`
`em` беруться від найближчого батьківського елемента

bachynskyi commented on src/app/task-js/task-js.component.ts:
```diff
@@ -0,0 +1,31 @@
+  ngOnInit() {
+    this.httpService.getData()
+      .subscribe((data: any) => {
+        this.Categories = data.categories;
```
Зауваження до назви. Змінні ми називаємо з маленької літери. Назви класів - з великої

bachynskyi commented on src/app/task-js/task-js.component.ts:
```diff
@@ -0,0 +1,31 @@
+  checked = true;
+
+  Categories: TestData[] = [];
+  Items: TestData[] = [];
```
В тебе `TestData` загальна модель для даних. Але категорії та айтеми мають різні проперті, тому і моделі мають бути різними

bachynskyi commented on src/app/task-js/task-js.component.ts:
```diff
@@ -0,0 +1,31 @@
+})
+export class TaskJSComponent implements OnInit {
+  cat = 1;
+  checked = true;
```
В тебе `checked` по дефолту `true`, а має бути значення з параметру `enable_multiple_lists`

bachynskyi commented on src/app/task-js/task-js.component.scss:
```diff
@@ -0,0 +1,38 @@
+    flex-direction: column;
+    padding: 20px;
+    margin: 20px;
+    background-color: aquamarine;
```
Колір в тебе має бути динамічним
